### PR TITLE
Fix getopt deprecation warnings (part 1)

### PR DESCRIPTION
**Pull request by edewata**
The following Python modules have been updated to use `argparse`:
* pki.server.cli.banner
* pki.server.cli.group
* pki.server.cli.jss
* pki.server.cli.listener
* pki.server.cli.migrate
* pki.server.cli.nss
* pki.server.cli.password

### CODE REVIEW

File: base/server/python/pki/server/cli/banner.py [hunk 1]
```diff
@@ -56,36 +71,19 @@ def usage(self):
...
-
-        except getopt.GetoptError as e:
-            logger.error(e)
-            self.usage()
-            sys.exit(1)
+        args = self.parser.parse_args(args=argv)
 
-        instance_name = 'pki-tomcat'
+        if args.help:
+            self.print_help()
```
Comment (fmarco76) [inline base/server/python/pki/server/cli/banner.py:77]: The method `print_help()` is not defined in the BannerShowCLI.

The argparse has its way to print the help message, we could use it making easier to keep option and description aligned but not sure how to configure with these modular commands. Can be investigated in separate PR

File: base/server/python/pki/server/cli/banner.py [hunk 2]
```diff
@@ -120,44 +140,21 @@ def usage(self):
...
-        instance_name = 'pki-tomcat'
-        banner_file = None
-        silent = False
+        args = self.parser.parse_args(args=argv)
 
-        for o, a in opts:
-            if o in ('-i', '--instance'):
-                instance_name = a
+        if args.help:
+            self.print_help()
```
Comment (fmarco76) [inline base/server/python/pki/server/cli/banner.py:146]: See above

**Review by fmarco76 — APPROVED** [review]
LGTM. Just a small error in `banner.py` but can be merged after the fix

File: base/server/python/pki/server/cli/banner.py (hunk 1 above)
Comment (edewata) [reply base/server/python/pki/server/cli/banner.py:77]: Good catch! I've renamed `usage()` to `print_help()`.

For now I just want to fix the deprecation warnings, but later we can try to use sub-parsers to define the options & help messages for the sub-commands.

**Comment by edewata** [comment]
@fmarco76 Thanks!
